OptimizelySDK/Event/BatchEventProcessor.cs

```diff
@@ -0,0 +1,346 @@
+﻿/* 
+ * Copyright 2019, Optimizely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using OptimizelySDK.Event.Entity;
+using System.Threading;
+using OptimizelySDK.Utils;
+using OptimizelySDK.Logger;
+using OptimizelySDK.ErrorHandler;
+using System.Linq;
+using OptimizelySDK.Event.Dispatcher;
+
+namespace OptimizelySDK.Event
+{
+    /**
+     * BatchEventProcessor is a batched implementation of the {@link EventProcessor}
+     *
+     * Events passed to the BatchEventProcessor are immediately added to a BlockingQueue.
+     *
+     * The BatchEventProcessor maintains a single consumer thread that pulls events off of
+     * the BlockingQueue and buffers them for either a configured batch size or for a
+     * maximum duration before the resulting LogEvent is sent to the NotificationManager.
+     */
+    public class BatchEventProcessor: EventProcessor, IDisposable
+    {
+        private const int DEFAULT_BATCH_SIZE = 10;
+        private static readonly TimeSpan DEFAULT_FLUSH_INTERVAL = TimeSpan.FromMinutes(1);
```

Comment:
I think I previously mentioned a default of 30 seconds. We should change it to that to be consistent.